zulipterminal/model.py

```diff
@@ -1907,6 +1907,15 @@ def _handle_realm_user_event(self, event: Event) -> None:
                     # realm_users has 'email' attribute and not 'new_email'
                     if "new_email" in updated_details:
                         realm_user["email"] = updated_details["new_email"]
+                    elif "custom_profile_field" in updated_details:
```

Comment:
This isn't accompanied by a change to the API, which was very confusing initially as mypy isn't complaining! I feel this is really a bug and lack of coverage in mypy, but the underlying issue appears to be since the TypedDict is defined in such a way that the keys are *definitely* present since it is 'total', so this branch isn't checked at all by mypy.

We have Message as total=False, but ideally we'd avoid that since the type-checking isn't as good. Would you be up for improving the typing of that?

Reply:
Since the realm user update event does not have "new_email" as a compulsory field, changing it to a ```NotRequired``` field in ```RealmUserEventPerson``` fixes this issue. Also, similar to "new_email", other fields are also not guaranteed to be present, except for ```user_id```. This means changing every field (except ```user_id```) to be a ```NotRequired``` field would also make sense.

Reply:
Yes, this is definitely a reasonable approach. If the different updates cannot be in the same event (eg. role and timezone?) then it would be stricter to use a Union of classes for each option. Otherwise NotRequired on lots of these indicates that somewhere between all or none of them could be present.